Generals/Code/GameEngine/Source/GameLogic/Object/Behavior/PropagandaTowerBehavior.cpp

```diff
@@ -205,11 +206,25 @@ UpdateSleepTime PropagandaTowerBehavior::update( void )
 		}
 	}
 
-	if( self->getContainedBy()  &&  self->getContainedBy()->getContainedBy() )
+#if RETAIL_COMPATIBLE_CRC
+	Bool contained = self->getContainedBy() && self->getContainedBy()->getContainedBy();
+#else
+	Bool contained = false;
+
+	for (Object* child = self, *container = self->getContainedBy(); container; child = container, container = container->getContainedBy())
+	{
+		ContainModuleInterface* containModule = container->getContain();
```

Comment:
How about we simplify this code here by creating a `Object::getEnclosingContain()`? which only returns `ContainModuleInterface*` if it is enclosing?

Reply:
Is the current approach unsatisfactory?

> Basically keep
> `getEnclosingContain` in line with `getContain`
> `getFirstEnclosingContainedBy` in line with `getContainedBy`

Functionally, `getEnclosingContain` is more in line with `getContainedBy` as both functions look up the containment chain for an external container object, whereas `getContain` is an instance accessor akin to `getAI`, `getPhysics`, `getStealth`, etc.

Reply:
I think I now understand your thinking behind `getEnclosingContain`, from natural language we can assume that it is a parent contain. My monkey brain things more of a "get the enclosing contain module that is owned by this object".

Reply:
I think my function would be clear with this name:
`ContainModuleInterface* Object::getContainIfEnclosing(Object* containee)`

Reply:
Yes. I suggest implement it roughly the way I suggested above. Can optimize the names where applicable.